src/HexagonOptimize.cpp

```diff
@@ -162,6 +162,8 @@ std::vector<Pattern> casts = {
     // Saturating narrowing casts
     { "halide.hexagon.satub.shr.vh.h", u8c(wild_i16x/wild_i16), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
     { "halide.hexagon.satuh.shr.vw.w", u16c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
+    { "halide.hexagon.sath.shr.vw.w", i16c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
+    { "halide.hexagon.satub.shr.vw.w", u8c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
```

Comment:
Yes, that is what i tried to do with deinterleave.vw.dv. It takes in a quad vector and moves the even words into a double vector (edv) and the odd words into another double vector (odv). Then I call "deinterleave.vw" on both edv and odv and then concat their results. Makes sense?